### PR TITLE
Compute coverage over sex chromosomes and predicted sex from a d4 file

### LINKED ISSUE #163: Compute sex according to sequencing data
Create a function that computes just that. This information goes into the coverage report so it must be avaiable.

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Fix #163 

This PR contains the code to compute coverage over X and Y and predict sex according to it.  Could have been a background function but **I've added also an endpoint to it**, so it can be used and tested by itself.

### How to test:
- give a real d4 file use the new endpoint to computer sex info about the sample

### Expected outcome:
- [x] The endpoint should return the expected stats

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested locally with path to a read d4 file: 🆗 

image

File: src/chanjo2/meta/handle_d4.py [hunk 1]
```diff
@@ -201,3 +202,34 @@ def get_gene_interval_coverage_completeness(
...
+            return PREDICTED_UNKNOWN
+        elif ratio <= 12:
+            # this is the entire prediction, it's usually very obvious
+            return PREDICTED_MALE
+        else:
+            # the few reads mapping to the Y chromosomes are artifacts
+            return PREDICTED_FEMALE
+
+
+def get_samples_sex_info(d4_file: D4File) -> Dict:
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:223]:
```suggestion
def get_samples_sex_metrics(d4_file: D4File) -> Dict:
```

File: src/chanjo2/meta/handle_d4.py [hunk 2]
```diff
@@ -201,3 +202,34 @@ def get_gene_interval_coverage_completeness(
             )
         )
     return intervals_cov
+
+
+def predict_sex(x_cov: float, y_cov: float) -> str:
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:207]: Would be great with some unit tests here for the different scenarios

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: src/chanjo2/constants.py [hunk 3]
```diff
@@ -101,3 +101,7 @@
         "Exon rank in transcript",
     ],
 }
+
+PREDICTED_MALE = "male"
+PREDICTED_FEMALE = "female"
+PREDICTED_UNKNOWN = "unknown"
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply src/chanjo2/constants.py]: Makes sense, done!

File: src/chanjo2/meta/handle_d4.py (hunk 2 above)
Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py:206]: Done!

**Pull request merged**